libs/cluster/Server/ClusterConfig.cs

```diff
@@ -516,17 +517,22 @@ public string GetNodeInfo(ushort workerId, ConnectionInfo info)
             //<config-epoch>
             //<link-state>
             //<slot> <slot> ... <slot>
-
-            return $"{workers[workerId].Nodeid} " +
-                $"{workers[workerId].Address}:{workers[workerId].Port}@{workers[workerId].Port + 10000},{workers[workerId].hostname} " +
-                $"{(workerId == 1 ? "myself," : "")}{(workers[workerId].Role == NodeRole.PRIMARY ? "master" : "slave")} " +
-                $"{(workers[workerId].Role == NodeRole.REPLICA ? workers[workerId].ReplicaOfNodeId : "-")} " +
-                $"{info.ping} " +
-                $"{info.pong} " +
-                $"{workers[workerId].ConfigEpoch} " +
-                $"{(info.connected || workerId == 1 ? "connected" : "disconnected")}" +
-                $"{GetSlotRange(workerId)}" +
-                $"{GetSpecialStates(workerId)}\n";
+            var nodeInfoStringBuilder = new StringBuilder();
+            return nodeInfoStringBuilder
+                .Append(workers[workerId].Nodeid).Append(" ")
+                .Append(workers[workerId].Address).Append(":").Append(workers[workerId].Port)
+                .Append("@").Append(workers[workerId].Port + 10000).Append(",").Append(workers[workerId].hostname).Append(" ")
+                .Append(workerId == 1 ? "myself," : "")
+                .Append(workers[workerId].Role == NodeRole.PRIMARY ? "master" : "slave").Append(" ")
+                .Append(workers[workerId].Role == NodeRole.REPLICA ? workers[workerId].ReplicaOfNodeId : "-").Append(" ")
+                .Append(info.ping).Append(" ")
+                .Append(info.pong).Append(" ")
+                .Append(workers[workerId].ConfigEpoch).Append(" ")
+                .Append(info.connected || workerId == 1 ? "connected" : "disconnected")
+                .Append(GetSlotRange(workerId))
```

Comment:
`GetSlotRange` and `GetSpecialStates` can return `StringBuilder` instances to avoid intermediate string allocation.

Reply:
i.e. we can call `GetSlotRange(workerId, nodeInfoStringBuilder)` here.